lib/lightning/invocation/dataclip.ex

```diff
@@ -51,9 +51,27 @@ defmodule Lightning.Invocation.Dataclip do
   def new(attrs \\ %{}) do
     change(%__MODULE__{id: Ecto.UUID.generate()}, attrs)
     |> change(attrs)
+    |> redact_password()
     |> validate()
   end
 
+  defp redact_password(
+         %Ecto.Changeset{
+           valid?: true,
+           changes: %{
+             body: %{"configuration" => %{"password" => _password}} = body
+           }
+         } =
+           changeset
+       ) do
+    body =
+      update_in(body, ["configuration", "password"], fn _any -> "***" end)
+
+    put_change(changeset, :body, body)
+  end
```

Comment:
Is this enough? We shouldn't be receiving the configuration key, @josephjclark the worker removes `configuration` from the outgoing dataclip right?

Either way, we used to (we definitely did it in platform) drop the entire `configuration` key since there are a huge variety of configuration schemas and `password` is one of many keys that can be sensitive.

Reply:
For the same scope of previous assertion it would be enough. Even if it's removed on the worker I propose keep it here and improve it later with your scrubber.